core/src/kura.rs

```diff
@@ -77,16 +71,26 @@ impl Kura {
         let kura = Arc::new(Self {
             mode,
             block_store: Mutex::new(Box::new(block_store)),
-            block_hash_array: Mutex::new(Vec::new()),
-            // broker,
-            block_reciever: Mutex::new(block_reciever),
-            block_sender,
+            block_data: Mutex::new(Vec::new()),
             block_plain_text_path,
         });
 
         Ok(kura)
     }
 
+    /// Create a kura instance that doesn't write to disk. Instead it serves as a handler
+    /// for in memory blocks only.
+    #[allow(clippy::unwrap)]
```

Comment:
Consider `#[cfg(test)]`.

Reply:
It doesn't work. Try it yourself and you'll see why.